### PR TITLE
Add a failing test for missing optional keywords

### PR DESCRIPTION
**Pull request by AtreyeeS**
**Description**

This pull request adds a failing test for a datastore with obs_index_table missing optional keyword `DEADC`
(see: #3352). Currently I mark this as xfail, which can be solved by adding `DEADC` as a required keyword.

We can either bring this up within the GADF or adapt gammapy to read the `DEADC` from the events header later

**Dear reviewer**

### CODE REVIEW

File: gammapy/data/tests/test_data_store.py [hunk 1]
```diff
@@ -132,6 +132,100 @@ def test_datastore_copy_obs_subset(tmp_path, data_store):
...
+    circle = CircleSkyRegion(
+        center=SkyCoord("83.63 deg", "22.14 deg"), radius=0.2 * u.deg
+    )
+    exclusion_mask = ~geom.region_mask(regions=[circle])
+    maker_fov = FoVBackgroundMaker(method="fit", exclusion_mask=exclusion_mask)
+
+    stacked = MapDataset.create(geom=geom, name="crab-stacked")
+    for obs in observations:
+        # First a cutout of the target map is produced
+        cutout = stacked.cutout(
```
Comment (adonath) [inline gammapy/data/tests/test_data_store.py:209]: Minor: maybe simplify and remove the cutout, I think performance wise it shouldn't really matter here...

File: gammapy/data/tests/test_data_store.py [hunk 2]
```diff
@@ -132,6 +132,100 @@ def test_datastore_copy_obs_subset(tmp_path, data_store):
...
+            obs.pointing_radec, width=2 * offset_max, name=f"obs-{obs.obs_id}"
+        )
+        # A MapDataset is filled in this cutout geometry
+        dataset = maker.run(cutout, obs)
+        # The data quality cut is applied
+        dataset = maker_safe_mask.run(dataset, obs)
+        # fit background model
+        dataset = maker_fov.run(dataset)
+        print(
```
Comment (adonath) [inline gammapy/data/tests/test_data_store.py:218]: Remove the print statement...

**Review by adonath — COMMENTED** [review]
Thanks a lot @AtreyeeS! I have two more proposals to simplify the test here:
- Maybe move the creation of the minimal data store to https://github.com/gammapy/gammapy-data/tree/master/tests and save the index files there
- Move the test to `gammapy/makers/tests/` and just access pre-defined data store.

**Comment by AtreyeeS** [comment]
Thanks @adonath ! Opened a PR in Gammapy-data and moved the test

**Review by adonath — APPROVED** [review]
Thanks a lot @AtreyeeS. no further comments from my side...

**Pull request merged**